java-base/Dockerfile

```diff
@@ -17,13 +17,20 @@ LABEL name="Stackable image for OpenJDK" \
     summary="The Stackable OpenJDK base image." \
     description="This image is the base image for all Stackable Java product images."
 
-# We need to use EPEL, as openjdk 22 is not shipped with UBI9
-RUN rpm --install --replacepkgs https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
+# See: https://adoptium.net/en-gb/installation/linux/#_centosrhelfedora_instructions
+RUN cat <<EOF > /etc/yum.repos.d/adoptium.repo
+[Adoptium]
+name=Adoptium
+baseurl=https://packages.adoptium.net/artifactory/rpm/rhel/\$releasever/\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.adoptium.net/artifactory/api/gpg/key/public
```

Comment:
note(non-blocking): Is this fine? We are always pulling the key from remote. Is this fine from a CRA (and friends) perspective?

CC @lfrancke

Reply:
or @dervoeti

Reply:
I'd say...for now....yes.

Reply:
Yeah, at some point we need to go through all remote resources we use during image builds and probably mirror them and figure out how to update them reliably and verify all signatures/hashes of everything that goes into a build... but today is not that day :slightly_smiling_face:
I look forward to that discussion though. Nix will be mentioned more than once.